### PR TITLE
Feat(scripts): improve npm run new to be wizard

### LINKED ISSUE #538: Improve npm run new to be wizard and off choices & pack
## User Story

- AS A Deverloper, site Builder, Tester
- I WANT a easier way to create starter/test site 
- SO THAT its easier to get a stand-alone test site.

### Acceptance Criteria

A wizard offer choices of

- Type of site : starter or test-site
- Asks where new site should be placed.
- Asks whether it should perform
  - gitinit
  - install
  - pack
  - overwrite existing site

### PR DESCRIPTION
**Pull request by hvanyo**
## Changes
Switch npm run new to wizard and add options.
Update test-site git ignore & package.json
Updated doc

## Test Instructions
Precursor step:  > npm run setup

1) Run npm run new with old parameters and all should work as before:
- npm run new ../startersite 
- npm run new ../startersite --noinstall
- npm run new ../testsite test-site 
- npm run new ../testsite test-site --noinstall

2) Run npm run new with NO parameters and then you can choose what you would like to do:

3) Starter - site

> npm run new

? What type of new site do you want to create? starter

? What is location of new site? recommend with path of ../ to be outside of monorepo.: ../starter-site

? Do you want to run git-init on Destination Directory? Recommended if New Starter Site: No. & Yes

? Do you want to run npm install on Destination Directory? (skip if you plan to Pack): No & Yes

? Do you want to pack latest version of Bodiless into Destination Directory? No & Yes

? Destination directory already exists. Do you want to stop? No & Yes

4) Test - site

> npm run new

? What type of new site do you want to create? test-site

? What is location of new site? recommend with path of ../ to be outside of monorepo.: ../test-site

? Do you want to run git-init on Destination Directory? Recommended if New Starter Site: No. & Yes

? Do you want to run npm install on Destination Directory? (skip if you plan to Pack): No & Yes

? Do you want to pack latest version of Bodiless into Destination Directory? No & Yes

? Destination directory already exists. Do you want to stop? No & Yes

## Related Issues

Fixes #538

### CODE REVIEW

File: examples/starter/README.md [hunk 1]
```diff
@@ -14,10 +14,9 @@ BodilessJS provides a Gatsby starter you can use as the basis of a new site.
 Currently, you must install it from the main repository as follows:
 
 ```bash
-git clone https://github.com/johnsonandjohnson/bodiless-js.git
-cd bodiless-js
-npm run ci
-npm run new /path/to/new/site
+git clone GIT_REPO NEW_SITE
+cd NEW_SITE 
+npm install
```
Comment (dewen) [inline examples/starter/README.md:19]: should this be `npm run setup`?

File: examples/starter/README.md [hunk 2]
```diff
@@ -14,10 +14,9 @@ BodilessJS provides a Gatsby starter you can use as the basis of a new site.
 Currently, you must install it from the main repository as follows:
 
 ```bash
-git clone https://github.com/johnsonandjohnson/bodiless-js.git
-cd bodiless-js
-npm run ci
-npm run new /path/to/new/site
+git clone GIT_REPO NEW_SITE
```
Comment (dewen) [inline examples/starter/README.md:17]: GIT_REPO referring BodilessJS repo?

File: packages/bodiless-migration-tool/doc/Technical/MigrationTool.md [hunk 3]
```diff
@@ -51,7 +51,7 @@ The tool is given URL that will crawl the existing site finding all pages, gener
 
     ```sh
     npm ci
-    npm run new ../NEW_MIGRATED_SITE
+    npm run new (follow wizard guidess from [here](/About/GettingStarted?id=creating-a-new-site))
```
Comment (dewen) [inline packages/bodiless-migration-tool/doc/Technical/MigrationTool.md:54]: guidess => guides

Comment (dewen) [reply packages/bodiless-migration-tool/doc/Technical/MigrationTool.md:54]: Shall we use full url?
https://johnsonandjohnson.github.io/Bodiless-JS/#/About/GettingStarted?id=creating-a-new-site

File: examples/starter/README.md (hunk 1 above)
Comment (dewen) [inline examples/starter/README.md:19]: Is GIT_REPO the `monorepo` here? If so shall we change to `npm run setup`?

File: scripts/new-starter.js [hunk 4]
```diff
@@ -13,48 +13,104 @@
...
 
-const args = process.argv.filter(arg => !arg.match(/^--/));
-const destDir = args[2] || path.resolve(home, 'gatsby-starter-bodiless');
-const src = args[3] || 'starter';
-const noInstall = process.argv.find(arg => arg === '--no-install');
-const srcDir = path.resolve('.', 'examples', src);
+var destDir = '';
+var src = '';
+var noInstall = '';
+var srcDir = '';
```
Comment (dewen) [inline scripts/new-starter.js:24]: const?

File: scripts/new-starter.js [hunk 5]
```diff
@@ -13,48 +13,104 @@
...
 
-try {
-  if (existsSync(destDir)) {
-    console.error(`${destDir} already exists. Stopping.`);
+
+let inquirer = require('inquirer');
```
Comment (dewen) [inline scripts/new-starter.js:27]: const?

File: scripts/new-starter.js [hunk 6]
```diff
@@ -13,48 +13,104 @@
...
 const { spawn } = require('child_process');
 const home = require('os').homedir();
 
-const args = process.argv.filter(arg => !arg.match(/^--/));
-const destDir = args[2] || path.resolve(home, 'gatsby-starter-bodiless');
-const src = args[3] || 'starter';
-const noInstall = process.argv.find(arg => arg === '--no-install');
-const srcDir = path.resolve('.', 'examples', src);
+var destDir = '';
+var src = '';
```
Comment (dewen) [inline scripts/new-starter.js:22]: let src default to `starter`?

File: scripts/new-starter.js [hunk 7]
```diff
@@ -13,48 +13,104 @@
...
+
+let inquirer = require('inquirer');
+
+const askSetup = () => {
+  const questions = [
+    {
+      type: 'list',
+      name: 'typeofsite',
+      message: 'What type of new site do you want to create?',
+      choices: ["starter", "test-site"],
```
Comment (dewen) [inline scripts/new-starter.js:35]: single quote for consistency.

File: scripts/new-starter.js [hunk 8]
```diff
@@ -13,48 +13,104 @@
...
+      default: src,
+    },
+    {
+      type: 'input',
+      name: 'destinationdir',
+      message: 'What is location of new site or site to update? recommend path to be ../ so its outside of monorepo.:'
```
Comment (dewen) [inline scripts/new-starter.js:41]: default to `destDir`?

File: examples/starter/README.md (hunk 1 above)
Comment (hvanyo) [reply examples/starter/README.md:19]: Whoops missed updating this one.

File: examples/test-site/package.json [hunk 9]
```diff
@@ -70,6 +70,7 @@
     "clean": "rimraf ./public/* ./.cache",
     "dev-backend": "node ./node_modules/@bodiless/backend/src/server.js",
     "dev-frontend": "gatsby develop -p 8005",
+    "init-psh": "bodiless-psh-init",
```
Comment (dewen) [inline examples/test-site/package.json:73]: do we need this script back?

It's described in examples/test-site/doc/Development/Platform.sh.md

Comment (hvanyo) [reply examples/test-site/package.json:73]: Its helpful when you want to update & test that platform.sh resource files are updated.

Comment (dewen) [reply examples/test-site/package.json:73]: ok

File: scripts/new-starter.js [hunk 10]
```diff
@@ -63,4 +115,62 @@ child.on('close', (code) => {
...
+  let noInstall = 0;
+
+  if (args.length >= 3) {
+    /// Preserve original new script behavior
+    destDir = args[2]
+    src = args[3] || 'starter';
+    noInstall = process.argv.find(arg => arg === '--no-install') || 1;
+    console.log(noInstall);
+  } else {
+    const choice = await askSetup();
```
Comment (dewen) [inline scripts/new-starter.js:146]: might want to use try/catch the error here.

**Review by dewen — DISMISSED** [review]

**Pull request closed without merging**